packages/flutter_tools/lib/src/commands/build_apk.dart

```diff
@@ -573,3 +581,15 @@ Future<int> buildApk(
 
   return result;
 }
+
+String _getLastBuildType(String buildDirectoryPath) {
+  File lastBuildFile = new File(path.join(buildDirectoryPath, '.build'));
```

Comment:
Done - changed to using a `build_meta.json` file with the entry `targetBuildType`.